### PR TITLE
[GPU] Add supporting blocked format for reverse operation

### PR DESCRIPTION
**Pull request by OlehKravchyshyn**
### Details:
 - added supporting blocked format to reverse operation
 - added unit test for verification new features

### Tickets:
 - 87569

### CODE REVIEW

File: src/plugins/intel_gpu/tests/test_cases/reverse_gpu_test.cpp [hunk 1]
```diff
@@ -47,31 +47,70 @@ struct reverse_gpu_test : public ::testing::TestWithParam<ReverseParams<T, mode>
...
         ReverseParams<T, mode> params = testing::TestWithParam<ReverseParams<T, mode>>::GetParam();
         auto& engine = get_test_engine();
-        tensor t;
 
-        auto reverse_input = engine.allocate_memory({data_type, params.input_format, params.input_tensor});
-        auto reverse_axes = engine.allocate_memory(
-            {ReverseModeTraits<mode>::data_type, format::bfyx, tensor(batch(1), feature(params.axis.size()))});
+        format fmt{format::bfyx};
+        bool reorder_needed = false;
+        switch (params.input_format) {
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/reverse_gpu_test.cpp:53]: I think it worth to extract this logic to separate function and maybe even move it to some common test utility

Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/tests/test_cases/reverse_gpu_test.cpp:53]: like
out = getPlainFormatFor(fmt);
reorderNeeded = out != fmtl

Comment (OlehKravchyshyn) [reply src/plugins/intel_gpu/tests/test_cases/reverse_gpu_test.cpp]: done

**Comment by wilson-seok** [comment]
No performance regression during int8 models.

**Review by wilson-seok — APPROVED** [review]

**Review by ahnyoung-paul — CHANGES_REQUESTED** [review]
Please rebase this PR.

**Review by ahnyoung-paul — APPROVED** [review]
LGTM

**Pull request merged**